cleanlab/object_detection/summary.py

```diff
@@ -444,6 +449,138 @@ def visualize(
+
+    predictions:
+        A list of ``N`` ``np.ndarray`` such that ``predictions[i]`` corresponds to the model predictions for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    num_procs:
+        Number of processes for parallelization. Default is 1.
+
+    Returns
+    -------
+        Average of TP,FP, FN for each class across iou thresholds.
+    """
+    iou_thrs = np.linspace(0.5, 0.95, int(np.round((0.95 - 0.5) / 0.05)) + 1, endpoint=True)
+    num_classes = len(predictions[0])
+    avg_metrics = {class_num: {"TP": 0.0, "FP": 0.0, "FN": 0.0} for class_num in range(num_classes)}
+
+    for iou_threshold in iou_thrs:
+        results_dict = _get_per_class_confusion_matrix_dict_(
+            labels, predictions, iou_threshold, num_procs
+        )
+
+        for class_num in results_dict:
+            tp = results_dict[class_num]["TP"]
+            fp = results_dict[class_num]["FP"]
+            fn = results_dict[class_num]["FN"]
+
+            avg_metrics[class_num]["TP"] += tp
+            avg_metrics[class_num]["FP"] += fp
+            avg_metrics[class_num]["FN"] += fn
+
+    num_thresholds = len(iou_thrs) * len(results_dict)
+
+    for class_num in avg_metrics:
+        avg_metrics[class_num]["TP"] /= num_thresholds
+        avg_metrics[class_num]["FP"] /= num_thresholds
+        avg_metrics[class_num]["FN"] /= num_thresholds
+    return avg_metrics
+
+
+def calculate_per_class_metrics(
+    labels: List[Dict[str, Any]], predictions: List[np.ndarray], num_procs: int = 1
+) -> Dict[int, Dict[str, float]]:
+    """
+    Parameters
+    ----------
+    labels:
+        A list of ``N`` dictionaries such that ``labels[i]`` contains the given labels for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    predictions:
+        A list of ``N`` ``np.ndarray`` such that ``predictions[i]`` corresponds to the model predictions for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    num_procs:
+        Number of processes for parallelization. Default is 1.
+
+    Returns
+    -------
+    per_class_metrics: dict
+        A dictionary containing per-class metrics.
+
+    """
+    avg_metrics = get_average_per_class_confusion_matrix_(labels, predictions, num_procs)
+
+    avg_metrics_dict = {}
+
+    for class_num in avg_metrics:
+        tp = avg_metrics[class_num]["TP"]
+        fp = avg_metrics[class_num]["FP"]
+        fn = avg_metrics[class_num]["FN"]
+
+        precision = tp / (tp + fp + 1e-12)  # Avoid division by zero
```

Comment:
There is  a `TINY_VALUE` variable we use for avoid division by zero like [here](https://github.com/cleanlab/cleanlab/blob/851831e643943966392285b134bb713ff6b78a75/cleanlab/object_detection/rank.py#L1118) if you can change to that instead

Reply:
added here: https://github.com/Steven-Yiran/cleanlab/blob/8fb478e87af9a5c27d215eb1b025e176c311143f/cleanlab/object_detection/summary.py#L590C1-L590C1